CalendarPicker/index.js

```diff
@@ -177,31 +177,111 @@ export default class CalendarPicker extends Component {
       onDateChange,
       enableDateChange,
       allowBackwardRangeSelect,
+      maxRangeDuration,
     } = this.props;
 
     if (!enableDateChange) {
       return;
     }
 
-    const date = moment({ year: currentYear, month: currentMonth, day, hour: 12 });
+    const date = moment({ year: currentYear, month: currentMonth, day, hour: 12 }),
+      startDate = selectedStartDate
+        ? moment(selectedStartDate)
+        : selectedStartDate,
+      endDate = selectedEndDate ? moment(selectedEndDate) : selectedEndDate;
```

Comment:
This could be simplified to:
```
startDate = moment(selectedStartDate);
endDate = moment(selectedEndDate);
```
If the date is already a moment object, it will result in a moment object.  If it's undefined, it'll run the same `moment(selectedStartDate)` logic in your ternary.